coders/composio_coders/swe.py

```diff
@@ -0,0 +1,232 @@
+  6. If you open a file and need to get to an area around a specific line that is not in the first 100 lines, say line 583, don't just use the scroll_down command multiple times. Instead, use the goto 583 command. It's much quicker.
+  7. If the bug reproduction script requires inputting/reading a specific file, such as buggy-input.png, and you'd like to understand how to input that file, conduct a search in the existing repo code, to see whether someone else has already done that. Do this by running the command: find_file "buggy-input.png" If that doesn't work, use the linux 'find' command.
+  8. Always make sure to look at the currently open file and the current working directory (which appears right after the currently open file). The currently open file might be in a different directory than the working directory! Note that some commands, such as 'create', open files, so they might change the current  open file.
+  9. When editing files, it is easy to accidentally specify a wrong line number or to write code with incorrect indentation. Always check the code after you issue an edit to make sure that it reflects what you wanted to accomplish. If it didn't, issue another command to fix it.
+  10. When you finish working on the issue, use submit patch tool to submit your patch.
+'''
+ISSUE_DESC_TMPL = '''
+ We're currently solving the following issue within our repository. Here's the issue text:
+    ISSUE_ID:
+    {issue_id}
+    ISSUE:
+    {issue}
+  Now, you're going to solve this issue on your own.
+  When you're satisfied with all of the changes you've made, you can submit your changes to the code base by simply running the submit command.
+  Note however that you cannot use any interactive session commands (e.g. python, vim) in this environment, but you can write scripts and run them. E.g. you can write a python script and then run it with `python </path/to/script>.py`.
+
+  If you are facing "module not found error", you can install dependencies. Example: in case error is "pandas not found", install pandas like this
+  `pip install pandas`
+'''
+
+curr_script_path = Path(__file__).resolve()
+script_dir = curr_script_path.parent
+LOGS_DIR_NAME_PREFIX = "coder_agent_logs"
+AGENT_LOGS_JSON_PATH = "agent_logs.json"
+
+
+def setup_logger():
+    handler = RichHandler(show_time=False, show_path=False)
+    handler.setLevel(logging.DEBUG)
+    logger = logging.getLogger("local_workspace")
+    logger.setLevel(logging.DEBUG)
+    logger.addHandler(handler)
+    logger.propagate = False
+    return logger
+
+
+logger = setup_logger()
+
+
+def get_git_root():
+    """Try and guess the git repo, since the conf.yml can be at the repo root"""
+    try:
+        repo = git.Repo(search_parent_directories=True)
+        return repo.working_tree_dir
+    except git.InvalidGitRepositoryError:
+        logger.info("no git repo found")
+        return None
```

Comment:
If we're only using `git` for this one function, I would suggest you use `subprocess` module  and read output for `git rev-parse --show-toplevel
` command, it will give you the same result so we can get rid of an extra dependency